### PR TITLE
Add some updated deps so hpi:run works

### PR DESCRIPTION
**Pull request by kzantow**
# Description

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by kzantow** [comment]
@michaelneale give this a clean build & hpi:run try

**Comment by michaelneale** [comment]
hey @vivek review please?

File: pom.xml [hunk 1]
```diff
@@ -544,6 +554,11 @@
             <version>1.86</version>
         </dependency>
         <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
```
Comment (vivek) [inline pom.xml]: Not sure if we need to add new dependency. looking at dependency:tree, 
```
[INFO] +- io.jenkins.blueocean:blueocean-web:jar:1.3.0-beta-1-SNAPSHOT:compile   
[INFO] |  +- org.jenkins-ci.plugins:metrics:jar:3.1.2.10:compile                 
[INFO] |  |  +- io.dropwizard.metrics:metrics-core:jar:3.1.2:compile             
[INFO] |  |  +- io.dropwizard.metrics:metrics-servlet:jar:3.1.2:compile          
[INFO] |  |  +- io.dropwizard.metrics:metrics-healthchecks:jar:3.1.2:compile     
[INFO] |  |  +- io.dropwizard.metrics:metrics-jvm:jar:3.1.2:compile              
[INFO] |  |  +- io.dropwizard.metrics:metrics-json:jar:3.1.2:compile             
[INFO] |  |  \- org.jenkins-ci.plugins:jackson2-api:jar:2.5.4:compile   
```

metrics plugin gets loaded before GitHub and thats causing older version to get resolved first by `maven hpi:run`. and then it fails to load GitHub api plugin as it needs newer version.

```
java.io.IOException: GitHub API Plugin v1.86 failed to load.
 - Jackson 2 API Plugin v2.5.4 is older than required. To fix, install v2.7.3 or later.
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:621)
	at hudson.PluginManager$2$1$1.run(PluginManager.java:517)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
```
Why not exclude jacson2-api version from metrics dependency?

**Comment by vivek** [comment]
Works fine. In one test I installed all suggested plugins and matrix authorization strategy failed to install but restart fixed it:

```
java.io.IOException: Failed to dynamically deploy this plugin
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1899)
	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1656)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to install matrix-auth plugin
	at hudson.PluginManager.dynamicLoad(PluginManager.java:875)
	at hudson.PluginManager.dynamicLoad(PluginManager.java:814)
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1895)
	... 5 more
Caused by: java.io.IOException: Matrix Authorization Strategy Plugin v1.7 failed to load.
 - Icon Shim Plugin v1.0.3 is older than required. To fix, install v2.0.3 or later.
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:621)
	at hudson.PluginManager.dynamicLoad(PluginManager.java:865)
	... 7 more
```

**Comment by kzantow** [comment]
Yes, a quick test shows this works fine for me, too. Seems like a bug in hpi:run, a bit... should find the most recent dependency required in the tree...

**Comment by michaelneale** [comment]
seems a bit better - still worrysome that it required a restart as the matrix version is newer, any idea what is pulling that in @vivek ? But if this passes - worth merging as it at leas tmakes it work.

**Comment by kzantow** [comment]
As noted on Gitter: hpi:run installs plugins, so a restart required due to version issues is a non-issue IMO

**Comment by michaelneale** [comment]
@kzantow a few failures in there that may be related (some are timing though). 

actionsTest – io.jenkins.blueocean.service.embedded.PipelineApiTest
a few seconds

File: blueocean-pipeline-api-impl/pom.xml [hunk 2]
```diff
@@ -101,6 +101,10 @@
             <groupId>org.jenkinsci.plugins</groupId>
             <artifactId>pipeline-model-definition</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
```
Comment (michaelneale) [inline blueocean-pipeline-api-impl/pom.xml]: did you try it with just this dependency? as that has helped before

**Comment by vivek** [comment]
>still worrysome that it required a restart as the matrix version is newer, any idea what is pulling that in @vivek ? But if this passes - worth merging as it at leas tmakes it work.

`blueocean-rest-impl` has matrix-auth test dependency `org.jenkins-ci.plugins.icon-shim:icon-shim:jar:2.0.3:test` but model definition gets loaded first and it loads `org.jenkins-ci.plugins.icon-shim:icon-shim:jar:1.0.3:compile`. This causes that error.

**Comment by michaelneale** [comment]
no there is some :bug: introduced with this - the tests are reliably failing (at least the unit tests) around input and actions

**Comment by michaelneale** [comment]
looks like they are not functional errors - the actions one - well there is one less action in the response - so what. The submit one - that is returning steps in a different order for no reason... so probably just tests written on changing assumptions... looking at correcting...

**Comment by michaelneale** [comment]
I can't even open blue ocean now with this... trying with a clean work...

**Comment by michaelneale** [comment]
@kzantow @vivek https://github.com/jenkinsci/blueocean-plugin/commit/23de5da216bcb0be2b878729cf50ddf7edb35274 fixes these tests - but I am not sure if these changes are papering over actual problems. 

Not sure why this behvaior changes, there must be other libraries that are updated transitively here? I would like this to be manually tested again, as I am not having a good time still getting hpi:run working - I am not sure if it is my machine though.

**Comment by kzantow** [comment]
Superseded by: https://github.com/jenkinsci/blueocean-plugin/pull/1360

**Pull request closed without merging**